[Christopher Tan] iP

chrisjohntan opened the pull request:
# Cloud: Lightweight tasker

>“Quit, don't quit... Noodles, don't noodles... You are too concerned about what was and what will be. There is a saying: yesterday is history, tomorrow is a mystery, but today is a gift. That is why it is called the present.” – Master Oogway

Cloud frees your mind of having to remember things you need to do. It's,
- 🌟 Intuitive and simple-to-use task manager 😹 
- Simple interface for fast input 🚀 🌚 
- Task categorization by todo, deadline, or event 📆  
- Supports saving and loading tasks from file 📂 

All you need to do is,
1. download it from [here](https://youtu.be/dQw4w9WgXcQ?si=knsS3DPp6RgsKU2z).
2. double-click it.
3. add your tasks ✍️ 
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Search for tasks
- [ ] Edit tasks (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chrisjohntan committed:
renamed chatbot to Cloud

added greet and exit

chrisjohntan committed:
echo user input, exit on command

chrisjohntan committed:
create TaskList for storing input tasks

chrisjohntan committed:
stop iterating tasks once reach head index

chrisjohntan committed:
increment head after adding new task

chrisjohntan committed:
newline after each task

chrisjohntan committed:
save each input to a TaskList

chrisjohntan committed:
move exit() out of loop

chrisjohntan committed:
Task class to represent tasks

chrisjohntan committed:
store tasks as an array of Task objects

chrisjohntan committed:
Return task status representation

chrisjohntan committed:
Update with the new TaskList impl

Check for mark and unmark commands
Mark and unmark tasks accordingly and print out the updates task status

chrisjohntan committed:
Update field access modifiers to protected

chrisjohntan committed:
Create ToDo class

chrisjohntan committed:
Create Deadline class

chrisjohntan committed:
Create Event class

chrisjohntan committed:
Rename Todo

chrisjohntan committed:
Create Event class

chrisjohntan committed:
Change array to ArrayList, add javadoc to some methods

chrisjohntan committed:
Use switch case to check commands

chrisjohntan committed:
Add Parser for parsing user inputs

chrisjohntan committed:
Create Query class for structuring user queries

chrisjohntan committed:
Edit string format

chrisjohntan committed:
Fix splitter

chrisjohntan committed:
Add overloaded constructor for queries with command only

chrisjohntan committed:
Handle adding tasks in a clsas method

chrisjohntan committed:
Rename to Todo.java

chrisjohntan committed:
Update file location

chrisjohntan committed:
Add testcases

chrisjohntan committed:
Add expected output

chrisjohntan committed:
Create exception classes

chrisjohntan committed:
Add command error checking

chrisjohntan committed:
Catch parsing errors

chrisjohntan committed:
Add delete task method

chrisjohntan committed:
Include delete command

chrisjohntan committed:
Add deleteTask method

chrisjohntan committed:
Create Storage class

chrisjohntan committed:
Add formatSave method for formatting to save file

chrisjohntan committed:
Add getTask method to get Task object at index

chrisjohntan committed:
Implement loading and saving of data to Storage

chrisjohntan committed:
Handle file errors internally

chrisjohntan committed:
Ignore /data/

chrisjohntan committed:
Create DateTime class

chrisjohntan committed:
Implement DateTime class for date management

chrisjohntan committed:
Fix exception handling

chrisjohntan committed:
Create Ui class to manage Ui I/O

chrisjohntan committed:
Use ui class to print to console

chrisjohntan committed:
Package classes

chrisjohntan committed:
Merge branch 'add-gradle-support'

chrisjohntan committed:
Fix date formatting

chrisjohntan committed:
Test DateTime class

chrisjohntan committed:
Test TaskList class

chrisjohntan committed:
Configure build for Jar

chrisjohntan committed:
Add javadoc header comments

chrisjohntan committed:
Reformat code to follow coding standard

chrisjohntan committed:
Add functionality for finding tasks by keyword

chrisjohntan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/cloud/Cloud.java
#	src/main/java/cloud/util/Parser.java

chrisjohntan committed:
Add checksyle config

chrisjohntan committed:
Fix style issues

ethan-goh commented on src/main/java/cloud/Cloud.java:
```diff
@@ -0,0 +1,166 @@
+
+public class Cloud {
+
+    private static final String EXIT_COMMAND = "bye";
```
Good following of Java coding practices here for constants.

ethan-goh commented on src/main/java/cloud/util/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+
+    /**
+     * Get the total number of tasks
+     *
+     * @return Count of all marked and unmarked tasks
+     */
+    public int getTaskCount() {
+        return tasks.size();
+    }
+
+    /**
+     * Get a task from the list
+     *
+     * @param index index of the task
+     * @return the requested task object
+     */
+    public Task getTask(int index) {
+        return this.tasks.get(index);
+    }
+
+    public TaskList find(String keyword) {
+        TaskList tasks = new TaskList();
+        for (Task task : this.tasks) {
+            if (task.getDescription().contains(keyword)) {
+                tasks.add(task);
+            }
+        }
+        return tasks;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        int nTasks = getTaskCount();
+        for (int index = 0; index < nTasks; index++) {
+            sb.append(String.format("%d: %s\n", index + 1, tasks.get(index)));
+        }
+        return sb.toString();
+    }
+}
```
Consistent use of JavaDocs, good job!

ethan-goh commented on src/main/java/cloud/util/Storage.java:
```diff
@@ -0,0 +1,94 @@
+package cloud.util;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import cloud.exception.CloudException;
+import cloud.task.Deadline;
+import cloud.task.Event;
+import cloud.task.Task;
+import cloud.task.Todo;
```
Import statements are consistent and organised

ethan-goh commented on src/main/java/cloud/util/Ui.java:
```diff
@@ -0,0 +1,57 @@
+        );
+    }
+
+    public void showExit() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    public void echo(String message) {
+        System.out.println(message);
+    }
+
+    public void showMarked(String taskStatus) {
+        System.out.println("cloud.task.Task marked as done!\n" + taskStatus);
+    }
+
+    public void showUnmarked(String taskStatus) {
+        System.out.println("cloud.task.Task marked as not done\n" + taskStatus);
+    }
+
+    public void showList(String list) {
+        System.out.println("Here is a list of all your tasks:\n" + list);
+    }
+
+    public void showMatching(String list) {
+        System.out.println("Here are the matching tasks in your list:\n" + list);
+    }
+
+    public void showError(String message) {
+        System.out.println(message);
+    }
+
```
Consistent naming of methods that adhere to Java coding standard.

ethan-goh reviewed (commented):
LGTM

Rachelcoll commented on src/main/java/cloud/task/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @Override
+    public String formatSave() {
```
Would it be better to name it as `getSaveFormat`?

Rachelcoll commented on src/main/java/cloud/util/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+     *
+     * @return String representation of the task
+     */
+    public String getLatestTask() {
```
Good job in following naming standard! 💯

Rachelcoll commented on src/test/java/cloud/util/TaskListTest.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Test
+    public void add_addTask_increaseTaskCount() {
```
Would it be better to rename it as "addTaskTest" according to the JUnit naming standard given?

Rachelcoll commented on src/main/java/cloud/Cloud.java:
```diff
@@ -0,0 +1,166 @@
+    private static final Storage storage = new Storage();
+    private static TaskList tasks = new TaskList();
+
+    private static void greet() {
+        System.out.println(
+                "Hello! I'm Cloud\n" +
+                        "What can I do for you?"
+        );
```
Good multiple line string indentation! 👍

Rachelcoll reviewed (commented):
LGTM, only some small nits to be fixed.

chrisjohntan committed:
Create command classes for executing commands

chrisjohntan committed:
Refactor parser to return Command objects

chrisjohntan committed:
Change datetime creation

chrisjohntan committed:
Change UI methods to return Strings

chrisjohntan committed:
Change deleteTask to return a Task object

chrisjohntan committed:
Add resources for javaFx

chrisjohntan committed:
Edit formatting

chrisjohntan committed:
Edit constructor to take in DateTime object

chrisjohntan committed:
Add javaFx

chrisjohntan committed:
Update build.gradle for javafx

chrisjohntan committed:
Create CloudException

chrisjohntan committed:
Add assertion to parse method

chrisjohntan committed:
Fix task format bug

chrisjohntan committed:
Add delimiter for task storage

chrisjohntan committed:
Storage.java: Improve code quality

The readData method does not follow SLAP, contains arrowhead nesting, and is too lengthy.

Improve code quality to make the code more readable.

Let's,
* refactor using extract method
* remove doubly nested try-catch blocks

chrisjohntan committed:
TaskList.java: Fix javadoc formatting and code quality

Some header comments do not follow the coding standard and some methods do not follow SLAP.

Let's
* ammend the header comments to follow coding standard
* use method extraction to refactor the offending methods

chrisjohntan committed:
Merge pull request #2 from chrisjohntan/branch-A-Assertions

Merge branch A-Assertions

chrisjohntan committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/cloud/util/Storage.java

chrisjohntan committed:
Merge pull request #3 from chrisjohntan/branch-A-CodeQuality

Merch branch A-CodeQuality

chrisjohntan committed:
Fix task index not incrementing

chrisjohntan committed:
Remove unused class

chrisjohntan committed:
Create help command to view the list of valid commands

chrisjohntan committed:
Remove unused Query class

chrisjohntan committed:
Improve javadoc quality and coverage

chrisjohntan committed:
Merge pull request #4 from chrisjohntan/Extension-C-Help

Merge Extension-C Help

chrisjohntan committed:
Create StorageException to handle Storage related errors

chrisjohntan committed:
Update user profile image

chrisjohntan committed:
Improve GUI design

GUI still has the default design from the tutorial. Profile images are too big, and there is not clear formatting for errors.

Tweak the GUI design to be cleaner and more minimalistic, improving user experience.

Let's,
* clean up the GUI design and styling
* add functionality to give error messages a different colour

chrisjohntan committed:
Add javadoc header comments and clean up code formatting

chrisjohntan committed:
Create CloudResponse class to improve organisation of responses

chrisjohntan committed:
Remove unused methods and change response format

chrisjohntan committed:
Update task find to be case insensitive

chrisjohntan committed:
Fix false incorrect date format bug

chrisjohntan committed:
Fix savefile formatting bug

chrisjohntan committed:
Handle task list index out of bounds error

chrisjohntan committed:
Update user guide and application screenshot

chrisjohntan committed:
Remove unused import

chrisjohntan committed:
Add quick start section to user guide
